src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -7364,8 +7361,8 @@ static void Test()
             CompileAndVerify(comp, expectedOutput: "in catch");
         }
 
-        [Fact]
-        public void PatternMatchReadonlySpanCharOnConstantString()
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void PatternMatchReadOnlySpanCharOnConstantString()
```

Comment:
1. I see we made a change to `ValueSetFactory` in original PR. Consider testing some of the behaviors related to our use of `EnumeratedValueSetFactory` the new scenarios (for example `is "a" and "b"`, multiple arms with duplicates, ...)
2. Do we have a test checking `GetTypeInfo` (Type and ConvertedType) on the string?
3. Along those lines, can we test `chars is "abc" and var x` (checking the type of `x`)
4. We should have a test with LangVer using RegularNext

Reply:
1 is covered; 2, 3 are included in the test plan; 4 is now covered by the `*CSharp10` tests.